Servers/WindowServer/Window.cpp

```diff
@@ -420,4 +420,19 @@ void Window::detach_client(Badge<ClientConnection>)
     m_client = nullptr;
 }
 
+void Window::recalculate_rect()
+{
+    if (is_resizable()) {
+        if (tiled() != WindowTileType::None) {
+            auto old_tiled = tiled();
+            set_tiled(WindowTileType::None);
+            set_tiled(old_tiled);
```

Comment:
This kind of incantation has a tendency to break later on.
Can we express this in a more direct way instead of "turn it off and on again"?